evap/student/templates/student_vote.html

```diff
@@ -309,42 +309,97 @@ <h3 class="mb-3">{{ evaluation.full_name }} ({{ evaluation.course.semester.name
+                const textareaAnswerFields = voteArea.querySelectorAll(".vote-inputs textarea");
                 const collapseToggle = voteArea.closest(".collapsible").querySelector(".collapse-toggle");
 
-                button.addEventListener("click", () => {
+                const clearContributorAnswers = () => {
                     voteArea.querySelectorAll(".vote-inputs [type=radio][value='6']").forEach(radioInput => {
                         radioInput.checked = true;
                         clearChoiceError(radioInput);
                     });
+                    textareaAnswerFields.forEach((textarea) => {
+                        textarea.value = "";
+                        textarea.dispatchEvent(new Event("input"));
+                    });
 
                     formSaver.saveAllData();
+                    // Disable this button, until user changes a value
+                    markNoAnswerButton.classList.remove("tab-selectable");
+                    markNoAnswerButton.disabled = true;
 
                     // hide questionnaire for contributor
                     const voteAreaCollapse = bootstrap.Collapse.getOrCreateInstance(voteArea);
                     voteAreaCollapse.hide();
                     collapseToggle.classList.add("tab-selectable");
 
-                    // Disable this button, until user changes a value
-                    button.classList.remove("tab-selectable");
-                    button.disabled = true;
+                    collapseEmptyTextanswerFields();
+                }
+                const collapseEmptyTextanswerFields = () => {
+                    textareaAnswerFields.forEach(textareaAnswerField => {
+                        if (textareaAnswerField.value.trim().length > 0) return;
+                        const parent = textareaAnswerField.closest('.row-question');
+                        console.log(parent);
+                        const button = parent.querySelector(".btn-textanswer");
+                        if (button === null) return;
+                        if (!button.classList.contains("collapsed")) {
+                            button.click();
+                        }
+                    });
+                };
+                markNoAnswerButton.addEventListener("click", () => {
+                    const shouldShowModal = () => {
+                        for (const textarea of textareaAnswerFields) {
+                            if (textarea.value.trim() !== "") {
+                                return true;
+                            }
+                        }
+
+                        const radioButtons = voteArea.querySelectorAll(".vote-inputs [type=radio]:not([value='6'])");
+                        for (const radioButton of radioButtons) {
+                            if (radioButton.checked) {
+                                return true;
+                            }
+                        }
+                        return false;
```

Comment:
I think we can do something like this:
```suggestion
                        return voteArea.querySelector(".vote-inputs [type=radio]:not([value='6']):checked") !== null;
```